content/guest-post/nextjs-authentication-guide/index.md

```diff
@@ -0,0 +1,388 @@
+
+    ```javascript
+    import { useEffect } from "react";
+    import { useLRAuth } from "loginradius-react";
+    import { Button, Heading, VStack, Center } from "@chakra-ui/react";
+
+    export default function Home() {
+
+    	const { loginWithRedirect, logout, isLoading, isAuthenticated, error, user } = useLRAuth();
+
+    	const router = useRouter();
+
+    	useEffect(() => {
+    		if (user && isAuthenticated) {
+
+    			router.push("/profile");
+
+    		}
+
+    	}, [router]);
+
+    	if (isLoading) {
+    		return <div>Loading...</div>;
+
+    	}
+
+
+    	if (error) {
+
+    		return <div>Oops... {error.message}</div>;
+
+    	}
+
+    	return (
+    		<Center pt={10}>
+    			<VStack spacing={[6, 8]}>
+    			<Heading as="h2">Welcome</Heading>
+
+    			{!user && (
+    				<Button onClick={() => loginWithRedirect()}>Login to continue</Button>
+    			)}
+
+    			{user && isAuthenticated &&  (
+    				<Button onClick={() => logout()}>Log out</Button>
+    			)}
+    		</Center>
+    	);
+    }
+    ```
+
+Let’s break down the above code snippet:
+
+- In the `useEffect`, it checks if there is a user object and if `isAuthenticated` is true. If these conditions are true, it redirects the user to their profile page. These conditions will only be true when the user has logged in. You'll set up the profile page later.
+- It checks if the page is loading using the `isLoading` boolean. If true, it displays “loading…”. In a production-ready application, you would display a loading spinner.
+- If there is an error during the authentication process, it accesses the message from the `error` object and displays it to the user.
+- Finally, you conditionally display the login and logout buttons with the information you get from `user` and `isAuthenticated`.
+
+### Displaying Authenticated User Data
+
+You have to display information about the authenticated user. Let’s set that up.
+
+    ```javascript
+    import { Box, Center, Heading } from "@chakra-ui/react";
+    import { useLRAuth } from "loginradius-react";
+    export default function Profile() {
+
+    const { isLoading, isAuthenticated, user } = useLRAuth();
+
+    	if (isLoading) {
+    		return <div>Loading...</div>;
+    	}
+
+    	if (isAuthenticated) {
+    		return (
+    			<Box>
+    				<Nav />
+    				<Center mt={10}>
+    					<Heading a="h2">
+    						Welcome to your profile {user.Email[0].Value}
+    					</Heading>
+    				</Center>
+    			</Box>
+    		);
+    	}
+    }
+    ```
+
+You should display a user’s information if they're authenticated. Here, you check that they're authenticated with the `isAuthenticated` hook. If they're authenticated, return some JSX and display the user’s email address.
+
+### Protecting a Route
+
+You can create protected routes to ensure only authenticated users can access those routes.
+
+    ```javascript
+    import { withAuthenticationRequired } from "loginradius-react"
```

Comment:
I would suggest using the entire code for Profile() as defined above for this section. You can add the `withAuthenticationRequired` part to that and highlight the changes.

Reply:
I don't understand. Do you mean I should combine sections 8 and 9 into a single section and explain everything together?

Reply:
Explained in the comment below